pkg/libvmi/vmi.go

```diff
@@ -107,14 +109,21 @@ func WithRng() Option {
 // WithWatchdog adds a watchdog to the vmi devices.
 func WithWatchdog(action v1.WatchdogAction) Option {
 	return func(vmi *v1.VirtualMachineInstance) {
-		vmi.Spec.Domain.Devices.Watchdog = &v1.Watchdog{
-			Name: "watchdog",
-			WatchdogDevice: v1.WatchdogDevice{
-				I6300ESB: &v1.I6300ESBWatchdog{
-					Action: action,
-				},
-			},
+
+		arch := runtime.GOARCH
```

Comment:
runtime pkg not required we can  use vmi.Spec.Arch to identify s390x refer other places (vmi is actual runtime obj)